pybamm/citations.py

```diff
@@ -216,9 +215,6 @@ def print(self, filename=None, output_format="text", verbose=False):
             Marquis2019 was cited due to the use of SPM
 
         """
-        # Parse citations that were not known keys at registration, but do not
-        # fail if they cannot be parsed
-        pybtex = have_optional_dependency("pybtex")
```

Comment:
You have two options here:

1. Either use a different function in the `Citations` class that calls `pybamm.have_optional_dependency()` in `test_util`—see my other comment—instead of `pybamm.print_citations()` (which is really the same as `pybamm.Citations.print()`), or
2. Re-introduce these deleted lines and check for the `bibtexparser` optional dependency before proceeding to the functionality of the method.

I think the latter is better in this case, since it allows for consistent API behaviour across PyBaMM versions despite internal changes in dependencies as they are being done here.

Reply:
pybtex was used here to obtain `text` output format as : 
```py
pybtex.format_from_strings(
                self._cited, style="plain", output_backend="plaintext"
            )
```
due to the lack of such functionality in `bibtexparser` we don't really have a reason to import it here and is left unused. Hence the failing pre-commit.

Reply:
We have to clear the style checks for the other PR checks to run. Is there a way you can use the unused variable?